ng-dev/utils/git/authenticated-git-client.ts

```diff
@@ -141,17 +141,27 @@ export class AuthenticatedGitClient extends GitClient {
     }));
   }
 
+  /** The previously configured access token. */
+  private static _token: string | null = null;
   /** The singleton instance of the `AuthenticatedGitClient`. */
-  private static _authenticatedInstance: AuthenticatedGitClient;
+  private static _authenticatedInstance: AuthenticatedGitClient | null = null;
```

Comment:
This should become a `Promise<AuthenticatedGitClient>` so that we don't have to create a new promise everytime right?

Reply:
I would need to think more about this. I considered it initially, but it required dealing with Promise chaining directly, which felt unexpected

Reply:
Creating the short-lived promise is not necessarily a problem here

Reply:
Done!